.github/workflows/test_suite.yml

```diff
@@ -0,0 +1,47 @@
+# Test suite workflow
+name: Test suite
+
+# Controls when the workflow will run
+on:
+
+  # Triggers the workflow on pushes to the 'main' branch
+  push:
+    branches: ["main"]
+
+  # Triggers the workflow on pushes to pull requests
```

Comment:
:+1: thanks for the detailed comments. It's clear to see what each bit of the workflow `yml` does